spec/controllers/queue_items_controller_spec.rb

```diff
@@ -155,45 +154,278 @@
+              expect(qi_1.reload.position).to eq(1)
+              expect(qi_2.reload.position).to eq(2)
+            end
+
+            it "sets a danger message" do
+              expect(flash[:danger]).to be_present
+            end
+
+            it "redirects back" do
+              expect(response).to redirect_to queue_path
+            end
+          end
+
+          context "when a position contains a non-digit character" do
+            let(:qi_1) { Fabricate(:queue_item, position: 1, user: pete) }
+            let(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+
+            before do
+              post :update, queue_items: { "#{qi_1.id}" => { position: "1123a", user_rating: 2 }, 
+                                           "#{qi_2.id}" => { position: 1.5, user_rating: 2 } 
+                                          }
+            end
+
+            it "does not update any queue items" do
+              expect(qi_1.reload.position).to eq(1)
+              expect(qi_2.reload.position).to eq(2)
+            end
+
+            it "sets a danger message" do
+              expect(flash[:danger]).to be_present
+            end
+
+            it "redirects back" do
+              expect(response).to redirect_to queue_path
+            end
+          end
+
+          context "when a position is set to a negative number" do
+            let(:qi_1) { Fabricate(:queue_item, position: 1, user: pete) }
+            let(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+
+            before do   
+              post :update, queue_items: { "#{qi_1.id}" => { position: 1, user_rating: 2 }, 
+                                           "#{qi_2.id}" => { position: -1, user_rating: 2 } 
+                                          }
+            end
+
+            it "does not update any queue items" do                              
+              expect(qi_1.reload.position).to eq(1)
+              expect(qi_2.reload.position).to eq(2)
+            end
+
+            it "sets a danger message" do
+              expect(flash[:danger]).to be_present
+            end
+
+            it "redirects back" do
+              expect(response).to redirect_to queue_path
+            end
+          end
+        end
+      end
+    
+      context "when user changes video rating and validations pass" do
+          let(:video1) { Fabricate(:video) }
+          let(:video2) { Fabricate(:video) }
+          let!(:qi_1) { Fabricate(:queue_item, position: 1, user: pete, video: video1) }
+          let!(:qi_2) { Fabricate(:queue_item, position: 2, user: pete) }
+          let!(:review1) { Fabricate(:review, user: pete, rating: 1, video: video1) }
+
+        it "updates the reviews if it already exists" do
+          post :update, queue_items: { "#{qi_1.id}" => { position: 1, user_rating: 4 } }
+          expect(qi_1.reload.user_rating).to eq(4)
+        end
+
+        it "creates a review with only the rating if it does not exist" do
```

Comment:
Your spec description says "with only the rating", but you do not verify that. Make sure that your expectations and descriptions match, or you could mislead someone reading your specs.